### ISSUE TITLE
confusing behavior in mlil expr_type

### ISSUE THREAD

**Issue by 3vilWind** [Vector35/binaryninja-api#2403]
**Describe the bug**
expr_type of MLIL_VAR differs from its src.type

**To Reproduce**
0. [test_case.zip](https://github.com/Vector35/binaryninja-api/files/6384136/test_case.zip)
1. Open test_case binary
2. Execute test_case.py content in console
3. It prints "void*" and "Test"

**Expected behavior**
It should print equal types "Test"

**Additional context**
Binary Ninja Version: 2.3.2763-dev Personal, 426a99ac

**Comment by plafosse** [comment]
I attempted to triage this issue for an hour and wasn't successful. The API I believe is correct. This issue will need further triage.

**Comment by 3vilWind** [comment]
![image](https://user-images.githubusercontent.com/29675029/134677210-c0c2b0dc-d91c-479d-8c13-11f1b2a1d891.png)
According to mlil graph, type of this expression should be "Type", not "void*" which expr_type returns

**Comment by plafosse** [comment]
Yes I agree what you're seeing doesn't seem correct. This issue will just need to be triaged by someone with more knowledge about the surrounding code.
